application/src/main/java/org/togetherjava/tjbot/commands/help/OnGuildLeaveCloseThreadListener.java

```diff
@@ -0,0 +1,72 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.ThreadChannel;
+import net.dv8tion.jda.api.events.GenericEvent;
+import net.dv8tion.jda.api.events.guild.member.GuildMemberRemoveEvent;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.EventReceiver;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.AddHelpChannel;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Remove all thread channels associated to a user when they leave the guild
+ */
+public class OnGuildLeaveCloseThreadListener implements EventReceiver {
+    private static final Logger logger =
+            LoggerFactory.getLogger(OnGuildLeaveCloseThreadListener.class);
+    private final Database database;
+
+    public OnGuildLeaveCloseThreadListener(@NotNull Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void onEvent(@NotNull GenericEvent event) {
+        if (event instanceof GuildMemberRemoveEvent leaveEvent) {
+            onGuildMemberRemoved(leaveEvent);
+        }
+    }
+
+    private void onGuildMemberRemoved(GuildMemberRemoveEvent leaveEvent) {
+        ThreadChannel threadChannel;
+        Set<Long> channelId = getThreadsAssociatedToLeaver(leaveEvent);
+        if (channelId.isEmpty()) {
+            logger.warn("Failed to get channelID user was associated with");
+        } else {
+            for (Long channel : channelId) {
```

Comment:
since this is a channel id the var name should say that
```suggestion
            for (Long channelId : channelIds) {
```

Reply:
> also y clear db after every 7 days y not remove each thread when it gets closed

I feel like adding a listener for every closed app would complicate the process. For example, if a user closes a thread and re-opens the thread via typing in it, I don't see an easy way in storing the data again.